packages/framework/aqueduct/src/data-objects/dataObject.ts

```diff
@@ -85,10 +85,11 @@ export abstract class DataObject<
 
 /**
  * Utility for creating SharedObjectKind instances for data objects.
+ * @typeParam T - {@link DataObjectKind}.
  * @internal
  */
-export function createDataObjectKind<T extends new (...any) => DataObject>(
+export function createDataObjectKind<T extends DataObjectKind>(
```

Comment:
`DataObjectKind` is more than just the constructor, right? Can we elaborate a bit on why this is the right thing?

Reply:
This change isn't strictly required but seems like a type-safety improvement. Maybe @ChumpChief could weigh in.

Reply:
I think this sounds fine?  But I get fuzzy when this much TS gets involved :-P. I'd trust @CraigMacomber primarily for any of the *Kind stuff.

Reply:
I wrote the original code here which is being replaced. It was incorrect, since DataObjects are actually build using their factory, and the constructor is only used for inferring the instance type. DataObjectKind captures both the factory, and the constructor (for typing).